container-engine-lib/lib/backend_impls/docker/docker_kurtosis_backend/backend_creator/backend_creator.go

```diff
@@ -219,9 +223,14 @@ func getDockerKurtosisBackend(
 		}
 
 		enclaveFreeIpAddrTrackers[enclaveUuid] = freeIpAddrProvider
+
+		serviceRegistrationRepository, err = service_registration.GetOrCreateNewServiceRegistrationRepository(enclaveDb)
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "An error occurred creating the service registration repository")
+		}
 	}
 
-	dockerKurtosisBackend := docker_kurtosis_backend.NewDockerKurtosisBackend(dockerManager, enclaveFreeIpAddrTrackers)
+	dockerKurtosisBackend := docker_kurtosis_backend.NewDockerKurtosisBackend(dockerManager, enclaveFreeIpAddrTrackers, serviceRegistrationRepository)
```

Comment:
I had to inject the repository because there are many state changes inside, for example, removing a service registration or updating the status.